cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -609,3 +613,45 @@ def test_guest_os_shutdown(appliance, provider, testing_vm_tools, ensure_vm_runn
         new_last_boot_time = view.entities.summary("Power Management").get_text_of("Last Boot Time")
         soft_assert(new_last_boot_time == last_boot_time,
                     "ui: {} should ==  orig: {}".format(new_last_boot_time, last_boot_time))
+
+
+@pytest.fixture(scope="function")
+def new_user(request, appliance):
+    user, user_data = _users(request, appliance, group="EvmGroup-vm_user")
+    yield appliance.collections.users.instantiate(
+        name=user[0].name,
+        credential=Credential(principal=user_data[0]["userid"], secret=user_data[0]["password"]),
+    )
+
+    if user[0].exists:
+        user[0].action.delete()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1687597])
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE, override=True)
+def test_retire_vm_with_vm_user_role(new_user, appliance, testing_vm):
+    """
+    Bugzilla:
+        1687597
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        startsin: 5.10
+        casecomponent: Automate
+        setup:
+            1. Provision vm
+        testSteps:
+            1. Create custom user with 'EvmRole_vm-user' role
+            2. Retire VM by log-in to custom user
+    """
+    # Log in with new user to retire the vm
+    with new_user:
+        new = testing_vm.find_quadicon(from_any_provider=True)
```

Comment:
@ganeshhubale I would suggest here to create `view` by using `get_entity` by `testing_vm.name` and use `check()`. Your approach is right but it will be restrict to Grid View and will not allow you to check above behavior for any other view. @digitronik please confirm this. :)

Reply:
Thanks @sbulage @digitronik